### PR TITLE
Fix NPE when init admin user

### PR DESCRIPTION
**Pull request by Linary**
Change-Id: I6fe0e341b5c6dfdb3d92b658cc13b470e8eb2a53

### CODE REVIEW

File: hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java [hunk 1]
```diff
@@ -79,7 +80,10 @@ public void setup(HugeConfig config) {
         String graphPath = config.getMap(ServerOptions.GRAPHS).get(graphName);
         E.checkArgument(graphPath != null,
                         "Invalid graph name '%s'", graphName);
-        this.graph = (HugeGraph) GraphFactory.open(graphPath);
+        HugeConfig hugeConfig = new HugeConfig(graphName);
+        // Forced set to false when initializing backend
+        hugeConfig.setProperty(CoreOptions.RAFT_MODE.name(), "false");
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java:85]: can't force set to false, since start-server also go here

**Review by javeme — CHANGES_REQUESTED** [review]

File: hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java [hunk 2]
```diff
@@ -79,7 +82,13 @@ public void setup(HugeConfig config) {
         String graphPath = config.getMap(ServerOptions.GRAPHS).get(graphName);
         E.checkArgument(graphPath != null,
                         "Invalid graph name '%s'", graphName);
-        this.graph = (HugeGraph) GraphFactory.open(graphPath);
+        HugeConfig hugeConfig = new HugeConfig(graphPath);
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java:85]: rename to graphConfig

File: hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java [hunk 3]
```diff
@@ -79,7 +82,13 @@ public void setup(HugeConfig config) {
...
+        if (config.getProperty(INITING_STORE) != null &&
+            config.getBoolean(INITING_STORE)) {
+            // Forced set to false when initializing backend
+            hugeConfig.setProperty(CoreOptions.RAFT_MODE.name(), "false");
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java:89]: "false" => false

**Comment by javeme** [comment]
also improve NPE code

File: hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java (hunk 3 above)
Comment (Linary) [reply hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java:89]: The option value must be String
```
private Object validateOption(String key, Object value) {
      E.checkArgument(value instanceof String,
                      "Invalid value for key '%s': %s", key, value);

      TypedOption<?, ?> option = OptionSpace.get(key);
      return option.parseConvert(value);
  }
```

File: hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java [hunk 4]
```diff
@@ -36,6 +37,8 @@
 
 public class StandardAuthenticator implements HugeAuthenticator {
 
+    private static final String INITING_STORE = "initing-store";
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java:40]: initing_store

File: hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java [hunk 5]
```diff
@@ -79,7 +82,13 @@ public void setup(HugeConfig config) {
         String graphPath = config.getMap(ServerOptions.GRAPHS).get(graphName);
         E.checkArgument(graphPath != null,
                         "Invalid graph name '%s'", graphName);
-        this.graph = (HugeGraph) GraphFactory.open(graphPath);
+        HugeConfig graphConfig = new HugeConfig(graphPath);
+        if (config.getProperty(INITING_STORE) != null &&
+            config.getBoolean(INITING_STORE)) {
+            // Forced set to false when initializing backend
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java:88]: Forced set RAFT_MODE to false

File: hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java [hunk 6]
```diff
@@ -79,7 +82,13 @@ public void setup(HugeConfig config) {
...
+            graphConfig.setProperty(CoreOptions.RAFT_MODE.name(), "false");
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java:89]: false

**Comment by Linary** [comment]
image

File: hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java [hunk 7]
```diff
@@ -79,7 +82,13 @@ public void setup(HugeConfig config) {
         String graphPath = config.getMap(ServerOptions.GRAPHS).get(graphName);
         E.checkArgument(graphPath != null,
                         "Invalid graph name '%s'", graphName);
-        this.graph = (HugeGraph) GraphFactory.open(graphPath);
+        HugeConfig graphConfig = new HugeConfig(graphPath);
+        if (config.getProperty(INITING_STORE) != null &&
+            config.getBoolean(INITING_STORE)) {
+            // Forced set RAFT_MODE to false
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java:88]: keep "when initializing backend"

**Review by javeme — APPROVED** [review]

**Review by zhoney — APPROVED** [review]

**Pull request merged**